### PR TITLE
autoyast: Fail with die and message to make errors obvious

### PR DESCRIPTION
**Pull request by okurz**
This change is [Reviewable](https://reviewable.io/reviews/os-autoinst/os-autoinst-distri-opensuse/1924)

### CODE REVIEW

File: tests/autoyast/installation.pm [hunk 1]
```diff
@@ -57,7 +57,6 @@ sub save_logs_and_continue {
 
 sub run {
     my $self = shift;
-    $self->result('ok');
```
Comment (coolo) [inline tests/autoyast/installation.pm]: this is still necessary - if you never fail

**Comment by coolo** [comment]
this is for @nadvornik to review

**Comment by coolo** [comment]
Vladimir?

**Comment by sysrich** [comment]
@nadvornik ping..?

**Comment by nadvornik** [comment]
LGTM, besides the issue mentioned by coolo

File: tests/autoyast/installation.pm (hunk 1 above)
Comment (okurz) [inline tests/autoyast/installation.pm]: I don't understand. Isn't that implicit? Why would we need to explicitly set this in this test?

Comment (coolo) [inline tests/autoyast/installation.pm]: the default result is na

Comment (okurz) [inline tests/autoyast/installation.pm]: ok, changed.

**Comment by okurz** [comment]
rebased and updated accordingly

**Pull request merged**